packages/core/src/components/forms/radioGroup.tsx

```diff
@@ -79,81 +78,85 @@ export interface RadioGroupProps extends Props, HTMLDivProps {
     selectedValue?: string | number;
 }
 
-let counter = 0;
-function nextName() {
-    return `${RadioGroup.displayName}-${counter++}`;
-}
-
 /**
  * Radio group component.
  *
  * @see https://blueprintjs.com/docs/#core/components/radio.radiogroup
  */
-export class RadioGroup extends AbstractPureComponent<RadioGroupProps> {
-    public static displayName = `${DISPLAYNAME_PREFIX}.RadioGroup`;
+export const RadioGroup: React.FC<RadioGroupProps> = props => {
+    const { children, className, disabled, inline, label, name, onChange, options, selectedValue, ...htmlProps } =
+        props;
 
     // a unique name for this group, which can be overridden by `name` prop.
-    private autoGroupName = nextName();
-
-    public render() {
-        const { disabled, label, options, className, children, name, onChange, ...htmlProps } = this.props;
-        const labelId = uniqueId("label");
-        return (
-            <div
-                role="radiogroup"
-                aria-labelledby={label ? labelId : undefined}
-                {...removeNonHTMLProps(htmlProps)}
-                className={classNames(Classes.RADIO_GROUP, className)}
-            >
-                {label && (
-                    <label className={Classes.LABEL} id={labelId}>
-                        {label}
-                    </label>
-                )}
-                {Array.isArray(options) ? this.renderOptions() : this.renderChildren()}
-            </div>
-        );
-    }
-
-    protected validateProps() {
-        if (this.props.children != null && this.props.options != null) {
+    const autoGroupName = React.useMemo(() => nextName(), []);
+
+    const labelId = React.useMemo(() => uniqueId("label"), []);
+
+    React.useEffect(() => {
+        if (children != null && options != null) {
```

Comment:
See my comment on spinner changes: https://github.com/palantir/blueprint/pull/7181/files#r1920669985

Reply:
Changed in 7da09cf6c4dbaf23dd53470a9bd33b8ca4ff6f84